include/kalmar_runtime.h

```diff
@@ -245,7 +264,7 @@ class KalmarQueue
   /// enqueue marker with prior dependency
   virtual std::shared_ptr<KalmarAsyncOp> EnqueueMarkerWithDependency(int count, std::shared_ptr <KalmarAsyncOp> *depOps, memory_scope scope) { return nullptr; }
```

Comment:
@bensander It seems the 3rd parameter, `scope` , is getting much more importance in this PR. Would you mind add more comments here to explain how `scope` works?

And looking at this PR I'm also wondering could this `scope` argument have a default value, say `hc::system_scope` ?